File: packages/snow_draw_engine/lib/draw/render/rect_intersection.dart
import '../types/draw_rect.dart';

/// Returns true when [a] and [b] overlap in world-space.
bool rectsIntersect(DrawRect a, DrawRect b) =>
    a.minX <= b.maxX &&
    a.maxX >= b.minX &&
    a.minY <= b.maxY &&
    a.maxY >= b.minY;

File: packages/snow_draw_rust_engine/src/draw/render/rect_intersection.rs
#![allow(dead_code)]

use crate::draw::types::draw_rect::DrawRect;

/// Returns `true` when two world-space rectangles overlap.
pub fn rects_intersect(a: DrawRect, b: DrawRect) -> bool {
    a.min_x <= b.max_x && a.max_x >= b.min_x && a.min_y <= b.max_y && a.max_y >= b.min_y
}
